src/voice-broadcast/audio/VoiceBroadcastRecording.ts

```diff
@@ -0,0 +1,120 @@
+}
+
+interface ChunkRecordedPayload {
+    buffer: Uint8Array;
+    length: number;
+    size: number;
+}
+
+/**
+ * This class provides the function to seamlessly record fixed length chunks.
+ * Subscribe with on(VoiceBroadcastRecordingEvents.ChunkRecorded, (payload: ChunkRecordedPayload) => {})
+ * to retrieve chunks while recording.
+ */
+export class VoiceBroadcastRecording implements IDestroyable {
+    private headers = new Uint8Array(0);
+    private chunkBuffer = new Uint8Array(0);
+    private previousChunkEndTimePosition = 0;
+    private pagesFromRecorderCount = 0;
+
+    public constructor(
+        private voiceRecording: VoiceRecording,
+        public readonly targetChunkLength: number,
+    ) {
+        this.voiceRecording.onDataAvailable = this.onDataAvailable;
+    }
+
+    public async start(): Promise<void> {
+        return this.voiceRecording.start();
+    }
+
+    public async stop(): Promise<void> {
+        await this.voiceRecording.stop();
+        this.emitAndResetChunk();
+    }
+
+    public on(event: string | symbol, listener: (...args: any[]) => void): this {
+        this.voiceRecording.on(event, listener);
+        return this;
+    }
+
+    public off(event: string | symbol, listener: (...args: any[]) => void): this {
+        this.voiceRecording.off(event, listener);
+        return this;
+    }
+
+    private onDataAvailable = (data: ArrayBuffer) => {
+        const dataArray = new Uint8Array(data);
+        this.pagesFromRecorderCount++;
+
+        if (this.pagesFromRecorderCount <= 2) {
+            // first two pages contain the headers
+            this.headers = concat(this.headers, dataArray);
+            return;
+        }
+
+        this.handleData(dataArray);
+    };
+
+    private handleData(data: Uint8Array) {
+        this.chunkBuffer = concat(this.chunkBuffer, data);
+        this.emitChunkIfTargetLengthReached();
+    }
+
+    private emitChunkIfTargetLengthReached() {
+        if (this.chunkLength >= this.targetChunkLength) {
+            this.emitAndResetChunk();
+        }
+    }
+
+    private emitAndResetChunk() {
+        if (this.chunkBuffer.length === 0) {
+            return;
+        }
+
+        const currentRecorderTime = this.voiceRecording.recorderSeconds;
+        const payload = {
+            buffer: concat(this.headers, this.chunkBuffer),
+            length: this.chunkLength,
+        } as ChunkRecordedPayload;
+        this.chunkBuffer = new Uint8Array(0);
+        this.previousChunkEndTimePosition = currentRecorderTime;
+        this.voiceRecording.emit(VoiceBroadcastRecordingEvents.ChunkRecorded, payload);
+    }
+
+    private get chunkLength() {
+        return this.voiceRecording.recorderSeconds - this.previousChunkEndTimePosition;
+    }
+
+    destroy(): void {
+        this.voiceRecording.destroy();
+    }
+}
+
+export const createVoiceBroadcastRecording = () => {
+    const targetChunkLength = SdkConfig.get("voice_broadcast")?.chunk_length;
```

Comment:
what should the default be? (when `chunk_length` is undefined)

Reply:
Config has a default. But yes, let's use the config default here, just in case